### PR TITLE
ScopedBuffer

### PR DESCRIPTION
**Pull request by volca02**
I found some unchecked memcpys in the dta file parser, and some freely positioned malloc/free constructs that like to leak easily. This makes things a bit more tight.

### CODE REVIEW

File: components/utils/scoped_buffer.hpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#ifndef UTILS_BUFFER_H
+#define UTILS_BUFFER_H
+
+#include <assert.h>
+#include <cstddef> // for size_t
+#include <string.h>
+#include <stdexcept>
+
+
```
Comment (drummyfish) [inline components/utils/scoped_buffer.hpp]: Should be put in MFUtil namespace.

File: components/utils/scoped_buffer.hpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+/// Non-Copyable (move passable) utilitary buffer with RAII scoping
+class ScopedBuffer {
```
Comment (drummyfish) [inline components/utils/scoped_buffer.hpp]: Bad bracket formatting, should be on a new line.

File: components/dta/parser.hpp [hunk 3]
```diff
@@ -229,7 +230,7 @@ bool DataFormatDTA::load(std::ifstream &srcFile)
     mFileTableRecords.clear();
 
     unsigned int headerArraySize = mFileHeader.mFileCount * sizeof(FileTableRecord);
-    FileTableRecord *headerArray = (FileTableRecord *) malloc(headerArraySize);
+    ScopedBuffer headerArray(headerArraySize);
```
Comment (drummyfish) [inline components/dta/parser.hpp]: We prefer fully-qualified names, i.e. MFUtil::ScopedBuffer (after adding that class to the namespace).

**Comment by drummyfish** [comment]
Nice, thank you for contribution. Just please take a look at the formatting, nothing big, just for consistency.

**Comment by volca02** [comment]
Hey thanks for the feedback. I have some muscle memory to battle here. Didn't notice you were using namespaces... I'll take a look at the comments and post a new merge request soon.

File: components/utils/scoped_buffer.hpp (hunk 1 above)
Comment (volca02) [reply components/utils/scoped_buffer.hpp]: I don't think it is generally a good idea to put include statements into namespaces - is that what you're proposing here?

Comment (volca02) [reply components/utils/scoped_buffer.hpp]: Oh you probably mean namespace should be started after the includes. Okay.

File: components/dta/parser.hpp (hunk 3 above)
Comment (volca02) [reply components/dta/parser.hpp]: Sure thing, makes sense. I'll put it into a merge request soon - I have a few more malloc statements removed throughout the codebase prepared to be commited, I'll splice these two edits together.

**Pull request merged**